src/main/java/core/basesyntax/Main.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+public class Main {
+    private static final int NUMBER_OF_FIGURES = 10;
+
+    public static void main(String[] args) {
+        FigureSupplier figureSupplier = new FigureSupplier();
+
+        Figure[] figures = new Figure[NUMBER_OF_FIGURES];
+        for (int i = 0; i < NUMBER_OF_FIGURES; i++) {
+            if (i < NUMBER_OF_FIGURES / 2) {
+                figures[i] = figureSupplier.getRandomFigure();
+            } else {
+                figures[i] = figureSupplier.getDefaultFigure();
+            }
+            figures[i].draw();
+        }
+    }
```

Comment:
According to the checklist, you should not create a new `FigureSupplier` inside the loop. However, it seems you have correctly created a single `FigureSupplier` instance before the loop starts, which is good practice.